includes/class-sensei-quiz.php

```diff
@@ -1634,6 +1635,52 @@ public static function the_quiz_pagination() {
 
 	}
 
+	/**
+	 * Rendering html element that will be replaced with Progress Bar.
+	 *
+	 * @since 3.15.0
+	 */
+	public static function the_quiz_progress_bar() {
+		$quiz_id             = get_the_ID();
+		$pagination_settings = json_decode(
+			get_post_meta( $quiz_id, '_pagination', true ),
+			true
+		);
+
+		if ( empty( $pagination_settings['pagination_number'] )
+			|| ! array_key_exists( 'show_progress_bar', $pagination_settings )
+			|| ! $pagination_settings['show_progress_bar'] ) {
+			return;
+		}
```

Comment:
I don't think this check is needed. There is another check a few lines down that is sufficient.

Also, keep in mind that there are cases where there is a `$pagination_settings['pagination_number']` but the pagination is not enabled. This happens when the `pagination_number` is bigger than the questions count.

Reply:
https://github.com/Automattic/sensei/pull/4606/commits/c0dccacdef11b3f6fbc123e5e350c888ca077291#diff-964a2368a54a4fd7dd7d8dfe593a8f74e690d028590dc23dbb8d1d4e5b68577fL1650